cpp/dolfinx/fem/Expression.h

```diff
@@ -238,7 +240,10 @@ class Expression
 
   /// @brief Evaluation points on the reference cell
   /// @return Evaluation points
-  const xt::xtensor<double, 2>& X() const { return _x_ref; }
+  std::pair<std::vector<double>, std::array<std::size_t, 2>> X() const
```

Comment:
Maybe add a comment saying that the evaluation points is returned as a flattened and the shape is returned as well. Another question is do we want to return a reference to the points (an std::span) instead of a copy?

Reply:
I think a copy is better. These are the reference points, so it shouldn't be called inside a loop. Returning a span or reference puts constraints on how the data is stored internally, and since it's not performance critical we don't need to have this constraint.